src/main/java/ogc/rs/apiserver/handlers/DxTokenAuthenticationHandler.java

```diff
@@ -148,9 +145,16 @@ public void handle(RoutingContext routingContext) {
                       routingContext.put(USER_KEY, user);
                       LOGGER.debug("the user key: " + routingContext.get(USER_KEY).toString());
                       routingContext.next();
-                  } else {
-                      LOGGER.debug("Authentication not successful: " + res.cause());
-                      routingContext.fail(new OgcException(401, "Invalid Token", res.cause().getMessage()));
+                  }
+                  else {
+                      if (isStacItemEndpoint && res.cause().getMessage().contains("token expired")) {
+                          LOGGER.debug("Ignoring token expiry for STAC item by ID endpoint.");
+                          routingContext.next();
+                      }
```

Comment:
If the token is expired, we don't need this special case for `isStacItemEndpoint`. We can just fail as usual with `Invalid Token`